Custom-Audio-Driver/Custom-Audio-Driver/OTDefaultAudioDevice-Mac.m

```diff
@@ -87,6 +87,31 @@ - (instancetype)init
         _safetyQueue = dispatch_queue_create("ot-audio-driver",
                                              DISPATCH_QUEUE_SERIAL);
         _restartRetryCount = 0;
+        
+        struct AudioObjectPropertyAddress devicePropertyAddress;
+        devicePropertyAddress.mSelector = kAudioHardwarePropertyDefaultOutputDevice;
+        devicePropertyAddress.mScope = kAudioObjectPropertyScopeGlobal;
+        devicePropertyAddress.mElement = kAudioObjectPropertyElementMain;
+        
+        AudioObjectPropertyListenerBlock audioObjectPropertyListenerBlock = ^(UInt32 numberAddresses, const AudioObjectPropertyAddress* addresses) {
+            NSLog(@"AudioObjectPropertyListenerBlock");
+            UInt32 index = 0;
+            while (index < numberAddresses) {
+                AudioObjectPropertyAddress address = addresses[index];
+                switch (address.mSelector) {
+                    case kAudioHardwarePropertyDefaultOutputDevice:
+                        [self setDefaultOutput];
+                        break;
+                    default:
+                        break;
+                }
+                index++;
+            }
+        };
+        
+        OSStatus status = AudioObjectAddPropertyListenerBlock(kAudioObjectSystemObject, &devicePropertyAddress, nil, audioObjectPropertyListenerBlock);
+        NSError *error = [NSError errorWithDomain:NSOSStatusErrorDomain code:status userInfo:nil];
```

Comment:
I don't understand why we are raising this error always. Probably it is my objc limitations the reason why I can't see

Reply:
Updated!